tests/tracer/test_agent.py

```diff
@@ -23,15 +23,82 @@ def test_stats_port(monkeypatch):
     assert agent.get_stats_port() == 1235
 
 
-def test_trace_url(monkeypatch):
-    assert agent.get_trace_url() == "http://localhost:8126"
-    monkeypatch.setenv("DD_TRACE_AGENT_PORT", "1235")
-    assert agent.get_trace_url() == "http://localhost:1235"
-    monkeypatch.setenv("DD_AGENT_HOST", "mars")
-    assert agent.get_trace_url() == "http://mars:1235"
+def test_trace_url(monkeypatch, mocker):
+    mock_exists = mocker.patch("os.path.exists")
 
-    monkeypatch.setenv("DD_TRACE_AGENT_URL", "http://saturn:1111")
-    assert agent.get_trace_url() == "http://saturn:1111"
+    # with nothing set by user, and the default UDS available, we choose UDS
+    mock_exists.return_value = True
+    assert agent.get_trace_url() == "unix:///var/run/datadog/apm.socket"
+    mock_exists.assert_called_once_with("/var/run/datadog/apm.socket")
+    mock_exists.reset_mock()
+
+    # with nothing set by user, and the default UDS unavailable, we choose default http address
+    mock_exists.return_value = False
+    assert agent.get_trace_url() == "http://localhost:8126"
+    mock_exists.assert_called_once_with("/var/run/datadog/apm.socket")
+    mock_exists.reset_mock()
+
+    # with port set by user, and default UDS unavailable, we choose user settings
+    with monkeypatch.context() as m:
+        m.setenv("DD_TRACE_AGENT_PORT", "1235")
+        assert agent.get_trace_url() == "http://localhost:1235"
+        mock_exists.assert_not_called()
+        mock_exists.reset()
```

Comment:
```suggestion
```

Reply:
if the mock wasn't called, we don't need to reset it.

I actually don't care much if we call reset after each scenario regardless, but in order to be consistent I would add `mock_exsts.reset()` after every scenario (there isn't consistency right now).